src/monmove.cpp

```diff
@@ -1381,6 +1527,73 @@ bool monster::attack_at( const tripoint &p )
     return false;
 }
 
+bool monster::pickup_at( const tripoint &p, item_location &target )
+{
+    //If called through normal pathfinding, then clear looting effect
+    if( p == goal ) {
+        remove_effect( effect_looting );
+
+    }
+
+    //Item has moved...
+    if( p != target.position() ) {
+        return false;
+    }
+
+    inv.push_back( *target.get_item() );
+    item *stored_item = &inv.back();
```

Comment:
There is no need to make this a pointer. A Reference would be better. Why do you even have this? It is used at *one* place only, and there you could just use `inv.back()` directly.